app/components/AsciidocBlocks/Listing.tsx

```diff
@@ -5,129 +5,36 @@
-        className: 'number',
-        match: /\d+[smhdw]/,
-        relevance: 0,
-      },
-      {
-        // 2024-05-27T00:00:00
-        className: 'number',
-        match: /@\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}/,
-        relevance: 0,
-      },
-      {
-        // @now()
-        className: 'number',
-        match: /@now\(\)/,
-        relevance: 0,
-      },
-      hljs.C_NUMBER_MODE,
-    ],
-  }
-})
-
-// Inspired by the HTML5 listing convert function
-// https://github.com/asciidoctor/asciidoctor/blob/82c5044d1ae5a45a83a8c82d26d5b5b86fcbc179/lib/asciidoctor/converter/html5.rb#L653-L678
-const Listing = ({ node }: { node: AdocTypes.Block }) => {
-  const document = node.getDocument()
-  const attrs = node.getAttributes()
-  const nowrap = node.isOption('nowrap') || !document.hasAttribute('prewrap')
-  const content = getContent(node)
-
-  // Replace callouts with placeholders
-  const replaceCallouts = (content: string) => {
-    const calloutRegex = /<i class="conum" data-value="\d+"><\/i>/g
-    const callouts: string[] = []
-    let placeholderContent = content.replace(calloutRegex, (match) => {
-      callouts.push(match)
-      return `__CALLOUT_PLACEHOLDER_${callouts.length - 1}__`
-    })
-    return { placeholderContent, callouts }
-  }
-
-  // Restore callouts from placeholders
-  const restoreCallouts = (highlightedContent: string, callouts: string[]) => {
-    return highlightedContent.replace(
-      /__CALLOUT_PLACEHOLDER_(\d+)__/g,
-      (_, index) => callouts[parseInt(index)],
-    )
-  }
+const Listing = ({ node }: { node: LiteralBlock }) => {
+  const { document } = useConverterContext()
 
-  // If we are highlighting we want the content to be decoded, since
-  // highlight.js will want the original content with tags included
-  // which it will then encode after highlighting.
-  // Otherwise let's just use the original content which comes from
-  // asciidoctor.js encoded.
-  const lang = attrs.language
+  const docAttrs = document.attributes || {}
+  const nowrap = node.attributes.nowrap || docAttrs['prewrap'] === undefined
 
-  const { placeholderContent, callouts } = replaceCallouts(content)
+  if (node.style === 'source') {
+    const lang = node.language
 
-  // Listing blocks of style `source` are source code, should have their syntax
-  // highlighted (where we have language support) and be inside both a `pre` and `code` tag
-  if (node.getStyle() === 'source') {
     return (
-      <div className="listingblock" {...getLineNumber(node)}>
-        <CaptionedTitle node={node} />
+      <div
+        className="listingblock"
+        {...(node.lineNumber ? { 'data-lineno': node.lineNumber } : {})}
+      >
+        <Title text={node.title} />
         <div className="content">
           <pre className={cn('highlight', nowrap ? ' nowrap' : '')}>
             {lang && lang === 'mermaid' ? (
-              <Mermaid content={decode(content)} />
+              <Mermaid content={node.source || ''} />
```

Comment:
We have a custom `Listing` converter here because we want to render the Mermaid elements. I tried moving this into the shared design system library, but it was unhappy with the Mermaid import even when it was set as a peer dep.